//////////////////////////////////////////////////////////////////////////
// 卡牌功能模块

HandleOpcode( CMSG_ALL_CHARACTER_INFO)

HandleOpcode( CMSG_SET_BATTLE_CHARACTER)

HandleOpcode(CMSG_GET_BATTLE_CHARACTER)

HandleOpcode( CMSG_USE_UPGRADE_LV_ITEM_TO_CHARACTER)
HandleOpcode( CMSG_UPGRADE_CHARACTER_QUALITY)

HandleOpcode( CMSG_INLAY_RUNE_TO_CHARACTER)
HandleOpcode( CMSG_UPGRADE_CHARACTER_REINFORCE)

HandleOpcode( CMSG_TRAINING_CHARACTER)
HandleOpcode( CMSG_TRAINING_CHARACTER_CONFIRM)
HandleOpcode( CMSG_UPGRADE_CHARACTER_SKILL)


HandleOpcode( CMSG_REBIRTH_CHAR)
HandleOpcode( CMSG_DECOMPOSE_CHAR)


HandleOpcode(CMSG_HERO_MOVE)
HandleOpcode(CMSG_CONSCRIPTION)

HandleOpcode(CMSG_STOP_HERO_MOVE)

HandleOpcode(CMSG_CHECK_PLAYER_ATTACH_INFO)
HandleOpcode(CMSG_BUY_HEADICON)
HandleOpcode(CMSG_SWITCH_HEADICON)

HandleOpcode(CMSG_QUIK_EXCHANGE_CHARACTER_EQUIPS)


/////////////排名

// HandleOpcode(CMSG_AUTO_ATTACK)//自动国战
// 
// HandleOpcode(CMSG_DONATE_CITY)//屯田区捐赠
// HandleOpcode(CMSG_TRAIN_SOLDIER)//练兵
// HandleOpcode(CMSG_BRIBE_CITY) //收买蛮族城市
// HandleOpcode(CMSG_INCITE_CITY)//发动蛮族城市

HandleOpcode(CMSG_REQUEST_DINNER)
HandleOpcode(CMSG_REQUEST_DINNER_DATA)

HandleOpcode(CMSG_PLAYER_VALUE_UPDATE)
HandleOpcode(CMSG_CHARACTER_UPDATE)